src/types/plugin-input.ts

```diff
@@ -13,7 +13,7 @@ export interface PluginInputs<T extends WebhookEventName = SupportedEvents> {
 
 export const permitRequestSchema = T.Object({
   type: T.Union([T.Literal("ERC20"), T.Literal("ERC721")]),
-  username: T.String(),
+  userId: T.Number(),
```

Comment:
Understandable. The reason I chose to use the id instead of the name is because it is consistent and immutable unlike the user name. For example it would break for me or @0x4007 that changed their user names recently, unlike our id that remains the same.

Reply:
Sorry for being late. My vision is to allow plugins (who can be developed by anybody) to pass in usernames. 

Under the hood, we should look up the ID during this transaction in the kernel (and use the ID in any code we control) 

This is because when the permit generation starts, the entire transaction takes less than a minute. The users involved in the task completion are unlikely to switch their username during the timeframe of the transaction so this will not pose any problems. 

I want plugin developers to be abstracted away from this implantation detail of always having to refer to users with their ID. 

Does that make sense?

Reply:
@0x4007 Yep totally makes sense. I was just worried about that scenario where users change their name, if that would break the logic of finding their ID afterwards. But all the previous references to the account get updated by GitHub on every pull request / issue etc. so this problem should not arise. I had reverted back to the `username` string instead of the ID so all good!